### PR TITLE
Fix NEXT-23457: Wrong creation of int number custom fields

### PR DESCRIPTION
**Pull request by flkasper**
Fix for ticket **NEXT-23457**: "Create Custom Fields in admin as type INT fields create it as FLOAT fields"

### 1. Why is this change necessary?

When a custom field is created in the admin, a int number field is saved as type float.

The error is in the merge of the CustomFieldConfig, the order is wrong. In the current code, the current (changed) config is overwritten with the default config, but this also means that config values are overwritten that already exist, In this case the config value of customFieldType.

### 2. What does this change do, exactly?

The sequence during the merge was fixed. During the merge the values of the defaultconfig are taken and merged with the current settings.

### 3. Describe each step to reproduce the issue or behaviour.

- Click "New custom field" to open modal for creating custom field
- Configure a customfield of type number with number type int
- Click add
- Custom field was created with float as number type, check by open edit modal

### 4. Please link to the relevant issues (if any).

fix [NEXT-23457](https://issues.shopware.com/issues/NEXT-23457)

### 5. Checklist

- [x] I have rebased my changes to remove merge conflicts
- [ ] I have written tests and verified that they fail without my change
- [x] I have created a [changelog file](https://github.com/shopware/platform/blob/trunk/adr/workflow/2020-08-03-implement-New-Changelog.md) with all necessary information about my changes
- [ ] I have written or adjusted the documentation according to my changes
- [ ] This change has comments for package types, values, functions, and non-obvious lines of code
- [x] I have read the contribution requirements and fulfil them.

### CODE REVIEW

File: src/Administration/Resources/app/administration/src/module/sw-settings-custom-field/component/sw-custom-field-detail/index.js [hunk 1]
```diff
@@ -165,8 +165,8 @@ Component.register('sw-custom-field-detail', {
             }
 
             this.currentCustomField.config = {
-                ...this.currentCustomField.config,
                 ...this.fieldTypes[customFieldType].config,
+                ...this.currentCustomField.config,
```
Comment (keulinho) [inline src/Administration/Resources/app/administration/src/module/sw-settings-custom-field/component/sw-custom-field-detail/index.js:169]: Can you add or adjust the existing test cases here: `Resources/app/administration/test/module/sw-settings-custom-field/component/sw-custom-field-detail.spec.js`

i'm wondering why it wasn't catched by the existing tests? Probably it is some edge case to the problem occurs only for some field types?

Comment (flkasper) [reply src/Administration/Resources/app/administration/src/module/sw-settings-custom-field/component/sw-custom-field-detail/index.js:169]: I can customize the test, it just takes a while since I haven't worked with Jest too much yet.

> i'm wondering why it wasn't catched by the existing tests? Probably it is some edge case to the problem occurs only for some field types?

The current test only checks the field types select & switch.
Furthermore, there are (currently!) only additional default configs (numberType) for the number type, which can also be changed afterwards by the corresponding editor component.

But it doesn't change the fact, that the order of the merge is wrong.
You always have to use the default values first, which will be overwritten with the current values.

Comment (flkasper) [reply src/Administration/Resources/app/administration/src/module/sw-settings-custom-field/component/sw-custom-field-detail/index.js:169]: I have started writing the test(s).

I first split the current test. The current test first started with type select, then switched to switch and then tested saving with it. The relevant method applyTypeConfiguration was also never executed there or tested separately.
So the test was neither complete nor correct before.

I started with appropriate adjustments, but currently come to my limits with Jest, as I do not have too much experience with it. Any help or guidance would be appreciated.

My problem:
After changing the customFieldType to number, the type specific editor component sw-custom-field-type-number is loaded with a select for the numberType (int|float).
Accessing the component is quite difficult, the find method only finds the select with a simple (and rather inaccurate) selector, but was eventually able to find it.
But when I change the value, this change is not applied.

Comment (keulinho) [reply src/Administration/Resources/app/administration/src/module/sw-settings-custom-field/component/sw-custom-field-detail/index.js:169]: Not entirely sure what's the issue there, but maybe that the change is applied async, maybe you can try `await Vue.nextTick();` as described here: https://v1.test-utils.vuejs.org/guides/testing-async-components.html

**Comment by keulinho** [comment]
Hey @fk-bitsandlikes i'm closing this PR for inactivity, feel free to reopen if you can follow up on the comments again.
